### ISSUE TITLE
Multiple <AnimatedOrphans/> required for animation (?)

### ISSUE THREAD

**Issue by nickschot** [ember-animation/ember-animated#169]
I'd like to animate the `finalRemoval` of an element which uses a `moveOver` type transition inside of an element which has `overflow: hidden` applied to it. The wanted effect is that it slides out of view within the context of this element.

The only way AFAIK that this is currently possible is by moving the `<AnimatedOrphans/>` component into this element so that the `overflow: hidden` of it also applies to any orphaned elements. This then of course causes problems for any other animations on the page that live outside of this element as there can currently be only 1 `<AnimatedOrphans/>` component.

I think the only solution is to have multiple (named) `<AnimatedOrphans/>` components present with a way to tell `removedSprites` where to go (with, I suppose, a default of the `<AnimatedOrphans/>` without a "name").

Any thoughts?

**Comment by ef4** [comment]
I was already considering allowing multiple AnimatedOrphans, in order to give people better control over the scope of when removed or inserted sprites should animate. For example, you could put AnimatedOrphans into your `posts` route and then it would handle transitions between `posts.index` and `posts.detail`, but would not handle transitions from `posts.index` to `about` (because it would not be present for the whole transition).

My preference for the API would be to have some form of "nearest" AnimatedOrphans handling each animator. Maybe that means AnimatedOrphans can become a block-form component, and it will take responsibility for its descendants. I like this better than explicit names because it would let you implement re-usable components that contain animators and can take advantage of whichever AnimatedOrphans they happen to live under.

**Comment by nickschot** [comment]
Yes that would indeed be a good way to approach it. I also like the block form best as that makes the definition of "nearest" AnimatedOrphans very clear, though it might not work for every possible situation (it does for mine, it's next to an outlet it could also be around).

I'll have a think too and look at the current implementation to see what's what.

**Comment by nickschot** [comment]
I'll take a stab at creating at least an initial implementation for this.
